test/e2e/network/kube_proxy.go

```diff
@@ -284,7 +284,9 @@ var _ = common.SIGDescribe("KubeProxy", func() {
 
 		// get proxyMode
 		stdout, err := e2epodoutput.RunHostCmd(fr.Namespace.Name, hostExecPodName, fmt.Sprintf("curl --silent 127.0.0.1:%d/proxyMode", ports.ProxyStatusPort))
-		framework.ExpectNoError(err)
+		if err != nil {
+			e2eskipper.Skipf("kube-proxy is not running or could not determine kube-proxy mode (%v)", err)
```

Comment:
we should not do dynamic skips, this test is meant to be used by kube-proxy only, openshift has to skip 

> var _ = common.SIGDescribe("KubeProxy", func() {

Reply:
oh, ok, it has the weird discovery thing

Reply:
/hold cancel

The test already does a lot of things we should not do, as to check the proxyMode, so is not going to be worse, I'm fine with this as an exception

/lgtm